src/typescript/sdk/src/indexer-v2/types/json-types.ts

```diff
@@ -125,6 +125,11 @@ type TransactionMetadata = {
   inserted_at?: PostgresTimestamp;
 };
 
+export type BlockAndEventIndexMetadata = {
+  block_number: Uint64String,
+  event_index: number,
```

Comment:
```suggestion
  block_number?: Uint64String,
  event_index?: number,
```

And then you'd do this, because the data is undefined at the individual field level here since it's the json type